### PR TITLE
TE-1430: add gomatic automation user

### PR DESCRIPTION
**Pull request by macdiesel**
@edx/pipeline-team PTAL.

### CODE REVIEW

File: playbooks/roles/go-server/tasks/main.yml [hunk 1]
```diff
@@ -123,6 +129,13 @@
     line: "{{ backup_user_password_line.stdout }}"
   when: GO_SERVER_BACKUP_USERNAME and GO_SERVER_BACKUP_PASSWORD
 
+- name: replace the gomatic line in the password file post-restore
```
Comment (feanil) [inline playbooks/roles/go-server/tasks/main.yml]: Now that we have 3 of these, we should convert this to a single task using with_items.

Comment (macdiesel) [inline playbooks/roles/go-server/tasks/main.yml]: done

**Comment by macdiesel** [comment]
@edx/pipeline-team comments addressed, PTAL.

File: playbooks/roles/go-server/tasks/main.yml [hunk 2]
```diff
@@ -109,19 +115,16 @@
...
-    regexp: "^{{ GO_SERVER_BACKUP_USERNAME }}"
-    line: "{{ backup_user_password_line.stdout }}"
-  when: GO_SERVER_BACKUP_USERNAME and GO_SERVER_BACKUP_PASSWORD
+    regexp: "^{{ item.username }}"
+    line: "{{ item.password_hash }}"
+  with_items:
+    - { username: GO_SERVER_ADMIN_USERNAME, password: GO_SERVER_ADMIN_PASSWORD, password_hash: admin_user_password_line.stdout }
+    - { username: GO_SERVER_BACKUP_USERNAME, password: GO_SERVER_BACKUP_PASSWORD, password_hash: backup_user_password_line.stdout }
+    - { username: GO_SERVER_GOMATIC_USERNAME, password: GO_SERVER_GOMATIC_PASSWORD, password_hash: gomatic_user_password_line.stdout }
+  when: item.username and item.password and item.password_hash
```
Comment (feanil) [inline playbooks/roles/go-server/tasks/main.yml:127]: Did you test this when condition if one of the stdouts is not generated?  I'm concerned that `x.stdout` will cause an error if `x` is not defined.

Comment (macdiesel) [inline playbooks/roles/go-server/tasks/main.yml:127]: good catch.  Done.

**Comment by feanil** [comment]
:+1:

File: playbooks/roles/go-server/tasks/main.yml [hunk 3]
```diff
@@ -87,6 +87,12 @@
   register: backup_user_password_line
   when: GO_SERVER_BACKUP_USERNAME and GO_SERVER_BACKUP_PASSWORD
 
+- name: generate line for go-server password file for gomatic user
+  command: >
+    /usr/bin/htpasswd -nbs "{{ GO_SERVER_GOMATIC_USERNAME }}" "{{ GO_SERVER_GOMATIC_PASSWORD }}"
+  register: gomatic_user_password_line
+  when: GO_SERVER_GOMATIC_USERNAME and GO_SERVER_GOMATIC_PASSWORD
```
Comment (doctoryes) [inline playbooks/roles/go-server/tasks/main.yml:94]: These new defs need to be defined as empty strings here, right?:
https://github.com/edx/configuration/blob/master/playbooks/roles/go-server/defaults/main.yml#L53

Comment (macdiesel) [inline playbooks/roles/go-server/tasks/main.yml:94]: Done

**Pull request merged**